### PR TITLE
Update and refresh company docs

### PR DESCRIPTION
**Pull request by stephaurelio**
### Proposed changes

This PR updates company docs based on support feedback:
- Callout with a tip for unique company names
- List prerequisites for creating companies on the overview page
- Update outdated FAQs
- Add more links to FAQs to surface them better from the overview pages
- Update the hierarchy diagram and description to explain that teams are optional

### Related issues (optional)

Closes JIRA https://docker.atlassian.net/browse/ENGDOCS-1811

### CODE REVIEW

File: content/docker-hub/company-faqs.md [hunk 1]
```diff
@@ -22,36 +22,39 @@ No, you can continue with business as usual.
...
 ### What features are supported at the company level?
 
-Domain verification, Single Sign-on, and System for Cross-domain Identity Management (SCIM) are supported at the company level. The following aren't supported:
+You can manage domain verification, Single Sign-on, and System for Cross-domain Identity Management (SCIM) at the company level. You can manage the following features at the organization level:
 
 - Image Access Management
 - Registry Access Management
 - User management
 - Billing
+- Domain audit
```
Comment (wmdane) [inline content/docker-hub/company-faqs.md:39]: Domain audit is not available for companies or organizations within a company.

Comment (stephaurelio) [reply content/docker-hub/company-faqs.md:39]: I saw that in the Slack thread. Will move this out of the list and change to: 
```suggestion
Domain audit is not supported for companies or organizations within a company.
```

File: content/admin/images/docker-admin-structure.png
Comment (vrunoa) [inline content/admin/images/docker-admin-structure.png:1]: @stephaurelio on this graph would we also want to add a direct line from `Organization` to `Member`. I know we have `Team (optional)` but maybe a direct line would make it more "visible"

Comment (stephaurelio) [reply content/admin/images/docker-admin-structure.png:1]: I like that idea, will update!

File: content/docker-hub/company-faqs.md [hunk 2]
```diff
@@ -22,36 +22,40 @@ No, you can continue with business as usual.
...
+### What permissions does the company owner have in the associated/nested organizations?
 
-Company owners can navigate to the **Organizations** page to view all their nested organizations in a single location. They can also view or edit organization members and change SSO and SCIM settings. Changes to company settings impact all users in each organization under the company.
+Company owners can navigate to the **Organizations** page to view all their nested organizations in a single location. They can also view or edit organization members and change SSO and SCIM settings. Changes to company settings impact all users in each organization under the company. See [Roles and permissions](../security/for-admins/roles-and-permissions.md).
 
 ### What features are supported at the company level?
 
-Domain verification, Single Sign-on, and System for Cross-domain Identity Management (SCIM) are supported at the company level. The following aren't supported:
+You can manage domain verification, Single Sign-on, and System for Cross-domain Identity Management (SCIM) at the company level. You can manage the following features at the organization level:
```
Comment (vrunoa) [inline content/docker-hub/company-faqs.md:33]: @stephaurelio would we still want to make visible that those features arent supported at the company level? 
```
The following features are not supported at the company level, but you can still manage them  at the organization level:
```

File: content/docker-hub/company-faqs.md [hunk 3]
```diff
@@ -22,36 +22,40 @@ No, you can continue with business as usual.
...
 
 - Image Access Management
 - Registry Access Management
 - User management
 - Billing
 
+You can manage users at the company level when you use Docker Admin.
```
Comment (vrunoa) [inline content/docker-hub/company-faqs.md:40]: @stephaurelio does this refer to User Management view? Would we want to be explicit here, i.e: "For helping you to view and manage users across all the organization under your company we're introducing User Management -> link to user management docs"

File: content/admin/_index.md [hunk 4]
```diff
@@ -24,7 +24,7 @@ The [Docker Admin](https://admin.docker.com) console provides administrators wit
   - Company owner: A company can have multiple owners. Company owners have company-wide observability and can manage company-wide settings that apply to all associated organizations. In addition, company owners have the same access as organization owners for all associated organizations.
 - Organization: An organization is a collection of teams and repositories. Docker Team and Business subscribers must have at least one organization.
   - Organization owner: An organization can have multiple owners. Organization owners have observability into their organization and can manage its users and settings.
-- Team: A team is a group of Docker members that belong to an organization. Organization and company owners can group members into additional teams to configure repository permissions on a per-team basis.
-  - Member: A member is a Docker user that's a member of at least one team in an organization.
+- Team (optional): A team is a group of Docker members that belong to an organization. Organization and company owners can group members into additional teams to configure repository permissions on a per-team basis. Using teams to group members is optional.
```
Comment (vrunoa) [inline content/admin/_index.md:27]: @stephaurelio does the "(optional)" needs to be explicit in the title? feels more like a description of an action itself like when adding member.

File: content/docker-hub/company-faqs.md (hunk 3 above)
Comment (stephaurelio) [reply content/docker-hub/company-faqs.md:40]: Yes, will update to link to the manage users with a company in Docker Admin docs. Thanks!

**Review by craig-osterhout — APPROVED** [review]
LGTM

**Pull request merged**